### PR TITLE
markdown: Add missing attribute to Markdown stub.

### PR DESCRIPTION
**Pull request by akshatdalton**
I'm a regular contributor @Zulip, and I've added a commit that overrides the [`_parseHeader` function](https://github.com/Python-Markdown/markdown/blob/3ea29f9d9b7d201269444dccedc0383065532b7e/markdown/extensions/codehilite.py#L164), and while working, I found that this attribute is missing here. (Related PR: https://github.com/zulip/zulip/pull/18608)

This attribute is defined here: https://github.com/Python-Markdown/markdown/blob/3ea29f9d9b7d201269444dccedc0383065532b7e/markdown/extensions/codehilite.py#L113.

### CODE REVIEW

**Comment by JelleZijlstra** [comment]
Your diff adds it as an argument to `__init__`, not an attribute. Looks like it should instead be an attribute of type `dict[str, Any]`.

**Comment by akshatdalton** [comment]
Added the changes in the new commit.
Force-pushed was for changing the commit message.
Please review.

**Comment by andersk** [comment]
Your diff still adds it as an argument to `__init__`, not an attribute.

**Comment by akshatdalton** [comment]
Yeah, right. I don't know how I forgot to change the main subject. :facepalm: 
Please review.

**Review by srittau — APPROVED** [review]

File: stubs/Markdown/markdown/extensions/codehilite.pyi [hunk 1]
```diff
@@ -18,6 +18,7 @@ class CodeHilite:
     tab_length: Any
     hl_lines: Any
     use_pygments: Any
+    options: Dict[str, Any]
     def __init__(
```
Comment (Akuli) [inline stubs/Markdown/markdown/extensions/codehilite.pyi:22]: Seems like we should also add `**kwargs: Any` to `__init__`. The source code looks like this:

```
    def __init__(self, src, **options):
        self.src = src
        self.lang = options.pop('lang', None)
        self.guess_lang = options.pop('guess_lang', True)
        ...
        self.options = options
```

Also, in `__init__`, it's missing `*` between `src` and `linenums` to mark the rest of the arguments as keyword-only.

Comment (akshatdalton) [reply stubs/Markdown/markdown/extensions/codehilite.pyi:22]: Good find!
I guess we only need to add `*` between `src` and `linenums`.

File: stubs/Markdown/markdown/extensions/codehilite.pyi [hunk 2]
```diff
@@ -18,9 +18,11 @@ class CodeHilite:
...
         self,
         src: Optional[Any] = ...,
+        *,
         linenums: Optional[Any] = ...,
         guess_lang: bool = ...,
         css_class: str = ...,
```
Comment (Akuli) [inline stubs/Markdown/markdown/extensions/codehilite.pyi:28]: It's still missing `**options: Any`, as you can pass in any options you want (and they go to `self.options`).

Comment (akshatdalton) [reply stubs/Markdown/markdown/extensions/codehilite.pyi:28]: Why do we want to add `**options: Any` when all the possible arguments that will be passed through `**options` are already defined here? :thinking: 
If there are some custom arguments that we want to pass, how will they be used inside this method?

Comment (akshatdalton) [reply stubs/Markdown/markdown/extensions/codehilite.pyi:28]: I've added the commit with this change.
Still, I'd be happy if you could answer the above question. :)

Comment (Akuli) [reply stubs/Markdown/markdown/extensions/codehilite.pyi:28]: Read the source code. It pops **some** things from the `options` dict, but everything that it doesn't pop goes to `self.options`.

**Review by Akuli — APPROVED** [review]

File: stubs/Markdown/markdown/extensions/codehilite.pyi (hunk 2 above)
Comment (akshatdalton) [reply stubs/Markdown/markdown/extensions/codehilite.pyi:28]: Okay, we can use the custom arguments using `self.options`.

**Pull request merged**